src/Components/Facility/InventoryLog.tsx

```diff
@@ -1,123 +1,90 @@
-import { useState, useCallback, useEffect, lazy } from "react";
+import { useState, lazy } from "react";
 
 import * as Notification from "../../Utils/Notifications.js";
-import { useDispatch } from "react-redux";
-import {
-  getInventoryLog,
-  flagInventoryItem,
-  deleteLastInventoryLog,
-  getAnyFacility,
-} from "../../Redux/actions";
-import { statusType, useAbortableEffect } from "../../Common/utils";
 import Pagination from "../Common/Pagination";
 import { formatDateTime } from "../../Utils/utils";
 import Page from "../Common/components/Page.js";
 import CareIcon from "../../CAREUI/icons/CareIcon.js";
 import ButtonV2 from "../Common/components/ButtonV2.js";
+import useQuery from "../../Utils/request/useQuery.js";
+import routes from "../../Redux/api.js";
+import request from "../../Utils/request/request.js";
 const Loading = lazy(() => import("../Common/Loading"));
 
 export default function InventoryLog(props: any) {
   const { facilityId, inventoryId }: any = props;
-
-  const dispatchAction: any = useDispatch();
-  const [isLoading, setIsLoading] = useState(false);
   const [saving, setSaving] = useState(false);
-  const initialInventory: any[] = [];
   let inventoryItem: any = null;
-  const [inventory, setInventory] = useState(initialInventory);
-  const [current_stock, setCurrentStock] = useState(0);
   const [offset, setOffset] = useState(0);
   const [currentPage, setCurrentPage] = useState(1);
-  const [totalCount, setTotalCount] = useState(0);
   const limit = 14;
   const item = inventoryId;
-  const [itemName, setItemName] = useState(" ");
-  const [facilityName, setFacilityName] = useState("");
 
-  const fetchData = useCallback(
-    async (status: statusType) => {
-      setIsLoading(true);
-      const res = await dispatchAction(
-        getInventoryLog(facilityId, { item, limit, offset })
-      );
-      if (!status.aborted) {
-        if (res?.data) {
-          setInventory(res.data.results);
-          setCurrentStock(res.data.results[0].current_stock);
-          setTotalCount(res.data.count);
-          setItemName(res.data.results[0].item_object.name);
-        }
-        setIsLoading(false);
-      }
-    },
-    [dispatchAction, offset, facilityId]
+  const { data: InventoryLog, refetch: InventoryFetch } = useQuery(
```

Comment:
It'd be cleaner to skip aliasing for primary queries of the components.

```suggestion
  const { data, refetch } = useQuery(
```